### PR TITLE
Update compilation documentation + Fix debugging for Microsoft OS

### PR DESCRIPTION
**Pull request by nmellado**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Doc update

* **What is the current behavior?** (You can also link to an open issue here)
Current build instruction described in documentation for MSVC+ninja do not work on VS 2019.

* **What is the new behavior (if this is a feature change)?**
Update doc

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Other information**:

### CODE REVIEW

**Comment by hoshiryu** [comment]
Could you provide instructions about how to debug a main-app plugin? 
I mean all the configuration/compilation/install procedure and what is needed beforehand from the windows dev toolkit.
Thanks

**Comment by nmellado** [comment]
Ok will do it.

**Comment by nmellado** [comment]
@hoshiryu the following is working on my side (using this PR), could you try it ?
 - Compile Radium-Engine in debug mode
 - Install it
 - Compile Radium-Apps in debug mode
 - Start `main-app.exe (install)`
image

**Comment by dlyr** [comment]
**CHECK LINEAR HISTORY FAIL**

Please rebase on current up to date master,
without merge commits (e.g. you can squash them).

**Comment by nmellado** [comment]
@hoshiryu I fixed a bug preventing to start the applications, and I added the installation of the pdb files in debug mode.
To test, you simply need to recompile and install Radium-Engine (don't need to reconfigure nor to empty the cmake cache).
Then you can recompile your application and should be able to debug: 

image

File: cmake/radium_setup_functions.cmake [hunk 1]
```diff
@@ -59,6 +59,7 @@ function(configure_cmdline_Radium_app)
         TARGETS ${ARGS_NAME}
         RUNTIME DESTINATION bin
     )
+
```
Comment (MathiasPaulin) [inline cmake/radium_setup_functions.cmake:62]:
```suggestion
```

File: cmake/radium_setup_functions.cmake [hunk 2]
```diff
@@ -128,6 +132,7 @@ function(configure_bundled_Radium_app)
         TARGETS ${ARGS_NAME}
         BUNDLE DESTINATION "bin/"
     )
+
```
Comment (MathiasPaulin) [inline cmake/radium_setup_functions.cmake:135]:
```suggestion
```

File: doc/basics/compilation.md [hunk 3]
```diff
@@ -167,34 +163,39 @@ To fix them, you need to edit the VS-specific file `CMakeSettings.json`, via *CM
...
+    }
+  ]
+}
 ~~~
 *Note*: it is strongly encouraged to use `/` separators in your path, instead of `\\` as previously mentionned. See https://stackoverflow.com/questions/13737370/cmake-error-invalid-escape-sequence-u
 
 ### Compilation
 
 Right click on CMakeList.txt > build > all.
+To install, you need to run any installation target, e.g. `HelloRadium.exe (install)`.
```
Comment (MathiasPaulin) [inline doc/basics/compilation.md:198]:
```suggestion
To install, you need to run any installation target, e.g. `Engine.dll (install)` or to select the menu <Build>/<Install radiumproject>
```

File: src/Engine/Renderer/Mesh/Mesh.hpp [hunk 4]
```diff
@@ -138,8 +138,13 @@ class RA_ENGINE_API AttribArrayDisplayable : public Displayable
...
         void operator()() {
-            m_displayable->m_dataDirty[m_idx] = true;
-            m_displayable->m_isDirty          = true;
+            if ( m_idx < m_displayable->m_dataDirty.size() )
+            {
+                m_displayable->m_dataDirty[m_idx] = true;
+                m_displayable->m_isDirty          = true;
+            }
+            else
+                LOG( logDEBUG ) << "Invalid dirty bit notified";
```
Comment (nmellado) [inline src/Engine/Renderer/Mesh/Mesh.hpp:147]:
```suggestion
                LOG( logDEBUG ) << "Invalid dirty bit notified"; /// \fixme Should never be here
```

**Pull request merged**